target/linux/ramips/patches-4.4/997-ralink-Introduce-fw_passed_dtb-to-arch-mips-ralink.patch

```diff
@@ -0,0 +1,28 @@
+--- a/arch/mips/ralink/of.c
++++ b/arch/mips/ralink/of.c
+@@ -81,13 +81,23 @@
+ 
+ void __init plat_mem_setup(void)
+ {
++	void *dtb;
++
+ 	set_io_port_base(KSEG1);
+ 
+ 	/*
+ 	 * Load the builtin devicetree. This causes the chosen node to be
+ 	 * parsed resulting in our memory appearing. fw_passed_dtb is used
+	 * by CONFIG_MIPS_APPENDED_RAW_DTB as well.
+ 	 */
+-	__dt_setup_arch(&__image_dtb);
++	if (fw_passed_dtb)
++		dtb = (void *)fw_passed_dtb;
++	else if (__dtb_start != __dtb_end)
++		dtb = (void *)__dtb_start;
++	else
++		dtb = &__image_dtb;
++
++	if (dtb)
++		__dt_setup_arch(dtb);
```

Comment:
patch is starting to look good, last comment is ... why is there an if in line 24 ? dtb should not be null and if it is __dt_setup_arch() will panic which is expected so i think it is safe to drop this if statement.